### ISSUE TITLE
Entity unavailable for indexed ID Exception

### ISSUE THREAD

**Issue by herriojr** [objectbox/objectbox-java#1150]
**Describe the bug**
It's hard to tell specifically where this is happening as I need to first obtain a stack trace first, however, we are getting a FileCorruptionException and it's complaining the "Entity unavailable for indexed ID 122" for some users. I don't understand exactly what this is and what I should be looking at to fix this issue.

**Basic info (please complete the following information):**
 - ObjectBox version (are you using the latest version?): 3.5.1
 - Reproducibility: Users seem to get stuck in this state
 - Device: Xiaomi Redmi A1+
 - OS: API 31

**To Reproduce**
1. No idea, wanted more details on this error and what can cause this state.

**Expected behavior**
No idea

**Code**
Cannot share

**Logs, stack traces**
Will provide errors when I am able to get them.

**Additional context**
- We use kotlin and we do use threads, however, writes are queued. Reads happen in parallel
- We do use subtransactions
- We did have some issue before that did require we call clearThreadResources() per another issue and try again (this was being called in retries in subtransactions that we will be fixing as this likely isn't safe)

**Comment by herriojr** [comment]
@greenrobot-team Does this happen to use any thread local caches when doing the lookup?

**Comment by greenrobot-team** [comment]
> any thread local caches

The cursor should not cache any data.

> retries in subtransactions

Also as I just realized this: the Java API will re-use the currently active transaction if a TX API is called inside an active transaction (e.g. check the `runInTx` code). So I'm not sure how retrying a "subtransaction" would even work (if I understand this correctly).

Regardless, one more info for us to determine the actual code path: is the affected `@Entity` using the default index on the String property, or is it using the `VALUE` (or `HASH64`) one? And as said, any additional info would help (logs or stack traces if you are able to obtain them).

**Comment by herriojr** [comment]
It wouldn't work which is why I corrected the original code to no longer do that (yet to be pushed to prod). I was trying to figure out if this could be the cause of the pre-described error.
